src/HotChocolate/Language/src/Language.SyntaxTree/ObjectValueNode.cs

```diff
@@ -28,10 +26,11 @@ public sealed class ObjectValueNode
         IReadOnlyList<ObjectFieldNode> fields)
     {
         Location = location;
-        Fields = fields ?? throw new ArgumentNullException(nameof(fields));
+        Fields = fields.OrderBy(t => t.Name.Value).ToArray() ??
```

Comment:
@michaelstaib, I have a question about this, is it legal to do? It was done to not do sorting all the time during equal and GetHashCode calls

Reply:
Yes, its illegal! 

- First, this is a syntax tree, it must exactly represent the parsed text.
- Second, we almost never use Linq since it performs badly in most cases. 
- Thrid, you just wasted memory by creating sorting and creating a new array.

Every change to the parser must be spec-compliant and is not allowed to have a negative performance impact.